src/utils/value_parsers.rs

```diff
@@ -10,3 +10,11 @@ pub fn kv_parser(s: &str) -> Result<(String, String)> {
         s[pos + 1..].parse().expect("infallible"),
     ))
 }
+
+/// Parse the value argument for a set metric. Floats are floored and strings are hashed.
+pub fn set_value_parser(s: &str) -> Result<f64> {
```

Comment:
Looks like this function can never return an error. If so, we should either return `f64` directly, or if we need a Result, we should return `Result<f64, Infallible>` to indicate that the function always succeeds. 

However, if we move to the enum for the metric value, maybe we will not need this function at all.